designdocs/passkey_p256.md

```diff
@@ -0,0 +1,101 @@
+# Design Doc for P256 PassKey Support
+
+## Table of Contents
+
+- [Design Doc for P256 PassKey Support](#design-doc-for-p256-passkey-support)
+  - [Table of Contents](#table-of-contents)
+  - [1. Introduction](#1-introduction)
+  - [2. Terminology](#2-terminology)
+    - [Accounts](#accounts)
+    - [Keys](#keys)
+    - [Signatures](#signatures)
+  - [3. Data Flow Diagram](#3-data-flow-diagram)
+  - [4. Data Map for Legal Teams](#4-data-map-for-legal-teams)
+  - [5. Specification](#5-specification)
+    - [Passkey Registration](#passkey-registration)
+    - [Security Considerations](#security-considerations)
+    - [Transaction Submission Specification](#transaction-submission-specification)
+  - [6. Implementation](#6-implementation)
+    - [Backend Example Code](#backend-example-code)
+    - [Frontend Example Code](#frontend-example-code)
+  - [7. Options for Discussion](#7-options-for-discussion)
+    - [Unsigned Extensions vs MultiSignature](#unsigned-extensions-vs-multisignature)
+    - [Generic Key Support](#generic-key-support)
+    - [Separate Pallet](#separate-pallet)
+  - [8. Conclusion](#8-conclusion)
+
+## 1. Introduction
+
+This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on Frequency chain. Passkey support aims to provide a novel non custodial solution for managing user accounts and signing transactions on-chain.
+
+## 2. Terminology
+
+### Accounts
+
+- **Authorized Account**: The user account associated with the Passkey.
+- **Parent Account**: The account holding the primary private key for transaction signing.
+- **Frequency Access**: The platform facilitating the management and interaction with user accounts.
+
+### Keys
+
+- **Passkey**: P256 key pair used for transaction signing and account management. This is the primary key used for transaction signing.
+- **Passkey Public Key**: The public key derived from the Passkey private key.
+- **Seed Phrase**: A mnemonic phrase used to generate cryptographic keys, particularly for SR25519 accounts. This is used for signing passkey proving account ownership.
```

Comment:
I recommend replace seed phrase with Frequency key so we would have 2 keys
1- Passkey
2- Frequency Account Key

Reply:
We might need some abbreviation for both, to make it easier

Reply:
For now i will call them `Account Keys` or `Account Public Key` so its generic